### DISCUSSION TITLE
How can I extract the link flow velocity during the simulation?

### DISCUSSION THREAD

**Discussion by QianyangWang** [pyswmm/pyswmm#425] [Q&A] [answered]
It seems the pyswmm.links.Link object does not have a property for obtaining the velocity value. I saw other people use their own velocity getter in their papers (StormReactor) to get the velocity, but I'm not sure about how to implement it. Thanks.

**Comment by bemcdonnell** [comment, marked as answer]
@QianyangWang, I looked in StormReactor and didn’t see the math for velocity.  However, you can get use Pyswmm get the flow rate and divide it by the average of the upstream and downstream flow areas. This should get you the average velocity for the pipe

**Reply by QianyangWang** [reply]
Thanks a lot! I've tried this solution, it seems the xs flow area has a different unit system (e.g. when the flow rate is 0.6 CMS while the xs flow area can be approximately 150, and the actual velocity in the SWMM simulation result is much higher than the calculated one). I'm not sure what kind of unit has been used by this variable. This phenomenon can even occur in the SWMM official Example-01.

**Reply by bemcdonnell** [reply]
You might like this information when thinking about units: https://stackoverflow.com/questions/74170188/how-to-access-units-programmatically-for-pyswmm

**Reply by QianyangWang** [reply]
Ah, I appreciate it.
